src/Sulu/Bundle/PageBundle/Controller/PageController.php

```diff
@@ -484,16 +480,20 @@ public function cgetAction(Request $request)
 
         $contents = [];
 
-        if ($parent) {
-            $contents = $contentRepository->findByParentUuid($parent, $locale, $webspaceKey, $mapping, $user);
-        } elseif ($webspaceKey) {
-            $contents = $contentRepository->findByWebspaceRoot($locale, $webspaceKey, $mapping, $user);
-        }
+        if ($ids) {
+            $contents = $contentRepository->findByUuids($ids, $locale, $mapping, $user);
```

Comment:
Instead of relying on the `nodeRepository` I am now relying on the `contentRepository` to load the pages. That way is ensured that the response looks the same, no matter if the `ids` parameter is passed or not (currently `admin/api/pages?flat=true` and `admin/api/pages?flat=true&ids=1,2,3` have a different structure).

Some of the fields are the same, and the only place I know this is used is the `page_selection`, which is still working after this change. I also have a similar problem for the snippets... The controller for snippets handles the localization different, it returns the an `originalLocale`, which is the one being requested and called `locale` e.g. for the pages`. Therefore it uses the `locale` property to return from which locale the data has been loaded, not which one was requested. I would like to make that consistent, because that is causing some bugs, although it is a BC break.

@sulu/core-team Do you see any problems with these changes?

Reply:
From my side I always thought:

`locale` = the requested locale
`originalLocale` = from where the data is loaded

Think we can change this to be consistent is it behave for pages and snippets different currently this is a bugfix for me we should definitly do.

Reply:
I dont know a lot about the implementation of snippets so i cannot really estimates the implications of this. But the way you describe it, it sound quite confusing at the moment. So if it makes things more consistent, I like it 🙂

Reply:
I just realized that there is a similar issue for the `single_selections`, which is not that easy to solve.

When a single page is requested from the API, it does not return the `locale` as part of the response. We also cannot simply add it, since it was possible until now to have a `locale` field in your template, which would then conflict, i.e. we have to get that in some different way. So we probably need to add a `router_attributes_to_view` configuration or something like that anyway, at least for the `single_page_selection`. Question is if we should handle that here in a similar way. Then it would not matter how the `locale` is returned from the API, although I would argue it would still make sense to do that in a more consistent way.

Reply:
I just omitted the `locale` in the configurations for the selections now, and this seems to be working as expected, because the sticky user locale mechanism @alexander-schranz built in #5101 kicks in. So strictly speaking this change would not be totally necessary. Should we still keep it and strive for consistency, or should I revert this change?